zstacklib/zstacklib/utils/lv_metadata.py

```
@@ -84,6 +107,66 @@ def _ensure_metadata_lv(self, metadata_path):
+        result = read_metadata(fence_path, lv_size)
+        if not result.is_usable():
+            raise MetadataIOError(
+                "cannot read metadata generation from fence LV %s: status=%s, error=%s" %
+                (fence_path, result.status, result.error))
+        try:
+            data = json.loads(result.payload)
+            if not data:
+                return 0
+            generation = int(data['generation'])
+        except (TypeError, ValueError, KeyError):
+            raise ValueError("invalid metadata generation in fence LV %s" % fence_path)
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)